rktlet: use seccompProfilePath for k8s 1.8

dongsupark opened the pull request:
As Kubernetes 1.8 changed its API to handle seccomp profiles inside `SecurityContext`, we need to change rktlet to support the new API.

Also update vendor to upgrade Kubernetes from 1.7.5 to 1.8.0, clean up the vendor tree by excluding test files as well as documents. Also fix build issues resulted from the k8s upgrade.

Fixes https://github.com/kubernetes-incubator/rktlet/issues/143

(NOTE: I tested this PR via kube-spawn on top of both k8s 1.7.5 and 1.8.0, and I could not find any particular issue, or any regression. Though we should be carefully test this PR, as it touches various parts.)

nhlfr commented on rktlet/runtime/helpers.go:
```diff
@@ -181,30 +180,23 @@ func getImageName(annotations map[string]string) string {
-func generateSeccompArg(annotations map[string]string, containerName string) (string, error) {
-	// by default kubernetes doesn't enable seccomp
+func generateSeccompArg(seccompProfile string, containerName string) (string, error) {
+	// by default kubernetes doesn't enable seccom
```
seccom -> seccomp

dongsupark replied:
@nhlfr Fixed the typo. Thanks!

iaguis commented:
I think we should fall back to checking the annotations if we can't find `SecurityContext.SeccompProfilePath`. I haven't tested it but I'm pretty sure this breaks seccomp profiles on v1.7.x clusters because we ignore the annotations.

iaguis commented:
However, the way it's implemented now we just set unconfined if there's no `SecurityContext.SeccompProfilePath` so it might be ok. 

Also, I checked and it seems other implementations differ on defaults: [cri-containerd](https://github.com/kubernetes-incubator/cri-containerd/blob/23b8330b443c12aba60e3998de3b24dcfc728ce0/pkg/server/container_create.go#L752) uses the `docker/default` profile and supports k8s 1.8 style,  [cri-o](https://github.com/kubernetes-incubator/cri-o/blob/db0bd66fca52179cf1dfb52b366145d59e5aa9a7/server/container_create.go#L1025) uses `unconfined` and supports k8s 1.7 style, and [dockershim](https://github.com/kubernetes/kubernetes/blob/c6a3f26988bce604c035dcd9024ce775ae4aafef/pkg/kubelet/dockershim/helpers.go#L61) uses `unconfined` and supports [k8s 1.8 style](https://github.com/kubernetes/kubernetes/blob/1f45cd06b31af32ef7eb074e6d962aecc193520d/pkg/kubelet/dockershim/docker_container.go#L149) (which is normal because it's inside the k8s repo).

dongsupark commented:
@iaguis Either way I'm fine, fallback to 1.7 or not.
BTW I discovered a regression w.r.t image names in this PR, so that every `rkt app add` fails with a strange error like `must give only one app`. I'm not sure why it happens, because the command-line options include only one app. 
So please do not merge this PR until we could find out its reason.

dongsupark commented:
It turns out, the failure `must give only one app` is a separate issue, which will be addressed via https://github.com/rkt/rkt/pull/3822.

Updated the PR to include fallback to the old annotations mechanism when there's no seccomp profile in SecurityContext.

dongsupark commented:
Updated.
* As reported by @iaguis, There were hidden build errors that were not recognized by running simple `make`. So I updated vendors including `k8s.io/api`, `k8s.io/apimachinery`, `k8s.io/client-go` etc. 
* Fixed build errors under `tests/framework`. Previously the tests were written based on k8s 1.6, while the current testing API of k8s 1.8 has changed a lot since then. Note that the test codes are still ugly, but I have just managed to only build without errors. Later we could improve the tests a little more.

iaguis commented on rktlet/runtime/helpers.go:
```diff
@@ -181,19 +181,25 @@ func getImageName(annotations map[string]string) string {
 
-	if profile == "unconfined" {
+	if profile == "unconfined" || profile == "" ||
+		strings.HasPrefix(profile, "localhost/") {
```
Also, this seems to test the wrong thing. I just tested this with:

```
  annotations:
      seccomp.security.alpha.kubernetes.io/pod: "localhost/test"
```

And I see:

```
2s          18s          3         security-context-demo.14f353277f693948   Pod       spec.containers{sec-ctx-demo}   Warning   Failed                             kubelet, kubespawn1      Error: seccomp profile "/var/lib/kubelet/seccomp/test" not supported
```

So it seems the kubelet resolves the right path and passes that to the rktlet.

iaguis replied:
I'd just remove the last predicate.

dongsupark replied:
@iaguis Done.
I rebased the tree, removing the check for localhost, and fixed another issue in a unit test.

iaguis commented:
This breaks tests in `tests/` (`make integ`) because you just add a fake runtime service.

I just checked and they're also broken on master (for different reasons), so I think we can leave it like it is in this PR and file an issue to fix them in the future.

dongsupark commented:
@iaguis Right, this PR just fixes build issues in the tests directory. It doesn't mean the tests work. For now I just don't have time to actually fix it. Hopefully we could do it later.
